src/ext/java/org/opentripplanner/ext/siri/updater/SiriSXUpdater.java

```diff
@@ -122,20 +123,23 @@ private void updateSiri() {
     } while (moreData);
   }
 
-  private Siri getUpdates() {
+  private Optional<Siri> getUpdates() {
     long t1 = System.currentTimeMillis();
     try {
-      Siri siri = siriHttpLoader.fetchSXFeed(requestorRef);
+      Optional<Siri> siri = siriHttpLoader.fetchSXFeed(requestorRef);
```

Comment:
Again, I would use flatMap() here which saves you dealing with the empty case: https://github.com/leonardehrenfried/OpenTripPlanner/commit/3f2e180e673ce9708b7ddb87215189e30c31a0c5